### DISCUSSION TITLE
Canary main file by @kyelmao

### DISCUSSION THREAD

**Discussion by pieckenst** [Obsidian-Development/helia#36] [General]
@kyelmao so im testing your main file rework and there were some problems with it

**Comment by pieckenst** [comment]
Like for example 
```
discord.Activity(type=discord.ActivityType.watching, name=f"{len(bot.guilds)} servers | {len(bot.shards)} shards!"),
```
this status got broken due to undeclared bot - and prevented the launching of the bot

**Reply by pieckenst** [reply]
And you forgot the : 
in your initial main file edit

**Comment by pieckenst** [comment]
and also https://github.com/Helia-Developers/helia/blob/canary/src/cogs/workers.py is no longer actually able to send request to bots.server-discord.com api due to being unable to get client id and number of guilds after this main file edit

**Reply by faluhub** [reply]
> and also https://github.com/Helia-Developers/helia/blob/canary/src/cogs/workers.py is no longer actually able to send request to bots.server-discord.com api due to being unable to get client id and number of guilds after this main file edit

when youre initializing the cog, you have self.bot so no need to add a bot thing in the function, just use self.bot

**Reply by pieckenst** [reply]
> 
> 
> > and also https://github.com/Helia-Developers/helia/blob/canary/src/cogs/workers.py is no longer actually able to send request to bots.server-discord.com api due to being unable to get client id and number of guilds after this main file edit
> 
> when youre initializing the cog, you have self.bot so no need to add a bot thing in the function, just use self.bot

Except that still results in same 
```
Debug information
Number of guilds:
0
Client ID:
Task exception was never retrieved
future: <Task finished name='Task-3' coro=<Workers.sdc_updater() done, defined at G:\helia-canary\src\cogs\workers.py:23> exception=AttributeError("'NoneType' object has no attribute 'id'")>
Traceback (most recent call last):
  File "G:\helia-canary\src\cogs\workers.py", line 33, in sdc_updater
    print(self.bot.user.id)
AttributeError: 'NoneType' object has no attribute 'id'
```
error

**Reply by pieckenst** [reply]
And that is supposed to show 129 servers
and the client id prior to sending request

**Reply by pieckenst** [reply]
@kyelmao

**Comment by faluhub** [comment]
> Like for example 
> ```
> discord.Activity(type=discord.ActivityType.watching, name=f"{len(bot.guilds)} servers | {len(bot.shards)} shards!"),
> ```
> this status got broken due to undeclared bot - and prevented the launching of the bot 

you need to replace 'bot' with 'self' since the class is the bot object

**Reply by pieckenst** [reply]
> 
> 
> > Like for example
> > ```
> > discord.Activity(type=discord.ActivityType.watching, name=f"{len(bot.guilds)} servers | {len(bot.shards)} shards!"),
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > this status got broken due to undeclared bot - and prevented the launching of the bot
> 
> you need to replace 'bot' with 'self' since the class is the bot object

This then shows number of shards and guilds as 0 instead of proper count

**Reply by pieckenst** [reply]
It is  supposed to show 129 guilds and 1 shard

**Reply by pieckenst** [reply]
@kyelmao

**Comment by faluhub** [comment]
all the things ive commented on, should be common sense and if you'd read the py docs and dpy docs more often, you could've fixed it urself. just pointing this out :)
